web/packages/teleterm/src/ui/HeadlessAuthn/HeadlessAuthentication.tsx

```diff
@@ -0,0 +1,73 @@
+/**
+ * Copyright 2023 Gravitational, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React, { useRef } from 'react';
+
+import { useAsync } from 'shared/hooks/useAsync';
+
+import { useAppContext } from 'teleterm/ui/appContextProvider';
+import { RootClusterUri } from 'teleterm/ui/uri';
+import { HeadlessAuthenticationState } from 'gen-proto-js/teleport/lib/teleterm/v1/service_pb';
+
+import { HeadlessPrompt } from './HeadlessPrompt';
+
+export function HeadlessAuthentication(props: HeadlessAuthenticationProps) {
+  const { headlessAuthenticationService, clustersService } = useAppContext();
+  const refAbortCtrl = useRef(clustersService.client.createAbortController());
+
+  const [updateHeadlessStateAttempt, updateHeadlessState] = useAsync(
+    async (state: HeadlessAuthenticationState) => {
+      return await headlessAuthenticationService.updateHeadlessAuthenticationState(
+        {
+          rootClusterURI: props.rootClusterURI,
+          headlessAuthenticationID: props.headlessAuthenticationID,
+          state: state,
+        },
+        refAbortCtrl.current.signal
+      );
+    }
+  );
```

Comment:
Nit: we don't need async/await here - we can pass the promise returned from `updateHeadlessAuthenticationState` directly to `useAsync`
```suggestion
  const [updateHeadlessStateAttempt, updateHeadlessState] = useAsync(
    (state: HeadlessAuthenticationState) =>
      headlessAuthenticationService.updateHeadlessAuthenticationState(
        {
          rootClusterURI: props.rootClusterURI,
          headlessAuthenticationID: props.headlessAuthenticationID,
          state: state,
        },
        refAbortCtrl.current.signal
      )
  );
```